### PR TITLE
[PB-685]: fix/trash-not-being-emptied-properly

### PR DESCRIPTION
**Pull request by miguelsw**
Issue:
- Did not completely delete the files

update:
- Set to count folders to delete

### CODE REVIEW

File: src/modules/trash/trash.usecase.ts [hunk 1]
```diff
@@ -20,10 +20,14 @@ export class TrashUseCases {
    * @param trashOwner User whose trash is going to be emptied
    */
   async emptyTrash(trashOwner: User): Promise<void> {
-    const count = await this.fileUseCases.getTrashFilesCount(trashOwner.id);
+    const filesCount = await this.fileUseCases.getTrashFilesCount(
+      trashOwner.id,
+    );
+    const foldersCount = await this.folderUseCases.getTrashFoldersCount(
```
Comment (sg-gs) [inline src/modules/trash/trash.usecase.ts:26]: It seems like this method does not filter already removed folders. Please do fix it including `{ removed: false }` on the query of folderUseCases.[getTrashFoldersCount](https://github.com/internxt/drive-server-wip/blob/ca4febf2c0978f1617c5c4dc94b0fe050c9a475c/src/modules/folder/folder.usecase.ts#L450)

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/modules/file/file.usecase.ts [hunk 2]
```diff
@@ -321,6 +321,8 @@ export class FileUseCases {
     return this.fileRepository.getFilesCountWhere({
       userId,
       status: FileStatus.TRASHED,
+      deleted: true,
```
Comment (sg-gs) [inline src/modules/file/file.usecase.ts:324]: Remove this, we are deprecating these fields for files

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/modules/file/file.usecase.ts [hunk 3]
```diff
@@ -321,6 +321,7 @@ export class FileUseCases {
     return this.fileRepository.getFilesCountWhere({
       userId,
       status: FileStatus.TRASHED,
+      removed: false,
```
Comment (sg-gs) [inline src/modules/file/file.usecase.ts:324]: Remove this field, is going to be deprecated, **only status should be used**

File: src/modules/trash/trash.usecase.ts [hunk 4]
```diff
@@ -33,10 +37,10 @@ export class TrashUseCases {
       await this.folderUseCases.deleteByUser(trashOwner, folders);
     }
 
-    for (let i = 0; i < count; i += emptyTrashChunkSize) {
+    for (let i = 0; i < filesCount; i += emptyTrashChunkSize) {
       const files = await this.fileUseCases.getFiles(
         trashOwner.id,
-        { deleted: true, removed: false },
+        { status: FileStatus.TRASHED, removed: false },
```
Comment (sg-gs) [inline src/modules/trash/trash.usecase.ts:43]: Remove the `removed` field usage, **only status should be used.**

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/modules/trash/trash.usecase.ts (hunk 1 above)
Comment (miguelsw) [reply src/modules/trash/trash.usecase.ts:26]: @sg-gs  it was updated in commits after this comment

**Review by sg-gs — APPROVED** [review]

**Pull request merged**